pkg/upgrade/control_plane.go

```diff
@@ -277,6 +278,10 @@ func (u *ControlPlaneUpgrader) updateMachine(name string, machine clusterv1.Mach
 		return err
 	}
 
+	if err := u.applyAnnotation(newMachine); err != nil {
```

Comment:
I found most errors to show up during machine deletion / etcd member removal, so I put the annotation before those steps so that resuming would work.

Reply:
Though, you do still end up with an orphaned machine in either case.

Reply:
:grimacing: should we track that as a separate issue?